cmds.go

```diff
@@ -49,6 +72,75 @@ type SonyCmdToggle struct {
 	Input string `short:"i" help:"Specify host input, do not autodetect"`
 }
 
+// AfterApply creates a new [Screen] from the flags in the [screenFlags] struct.
+func (sf *screenFlags) AfterApply() error {
+	s, err := NewScreen(sf.Display, sf.Manufacturer, sf.ProductCode)
+	if err != nil {
+		return err
+	}
+	sf.screen = s
+	return nil
+}
+
+// Run (offscreen run) runs offscreen to turn the connected TV on and off
+// in line with X screen saver events.
+func (cmd *RunCmd) Run() error {
+	ch := make(chan bool)
+	g := errgroup.Group{}
+	g.Go(func() error { return cmd.screen.Watch(ch) })
+	g.Go(func() error { return cmd.loop(ch) })
```

Comment:
I don't quite understand the need for channels here.

Reply:
no need, just the design. it was more complex with multiple event loops as I had the screen saver events and the randr events (monitor presence) separated. but I merged them. Perhaps you are right and an interface would make this simpler. I'll try that out